### ISSUE TITLE
Show matching pool in tallying phase cart

### ISSUE THREAD

**Issue by ryancreatescopy** [ethereum/clrfund#106]
Update: see latest designs at bottom. Given matching pool amount is always calculated (never editable), let's not display it as a "cart item" but show as line item before cart total.

![image](https://user-images.githubusercontent.com/40891631/120464969-68c1b200-c395-11eb-9aed-235bfd0bbcda.png)

If I reallocate and have to donate to the matching pool due to under allocation, a cart item showing matching pool should show in my contributions list...

![Image 2021-06-02 at 9 13 10 AM](https://user-images.githubusercontent.com/8097623/120515276-02cc2f00-c383-11eb-81e3-528ff1b292e5.jpg)

**Comment by samajammin** [comment]
@ryancreatescopy couple questions:

- Do we want to display this during reallocation phase for user as well?

- Do we display a cart item for the matching pool before the tx? Or only one that reallocation has been made?

**Comment by corwintines** [comment]
@samajammin @ryancreatescopy were we just wanting to show the contribution to the matching pool like we do in the edit cart view?
<img width="367" alt="Screen Shot 2021-06-03 at 3 42 17 PM" src="https://user-images.githubusercontent.com/15589226/120715772-51acbe00-c482-11eb-86c2-bfa91ee7b756.png">

If that makes sense. We would have the cart items for the projects that were contributed to, but could just show the matching pool contribution like we do in the edit cart for consistency. Or were you wanting a default cart item for the matching pool?

Im trying to figure out how the cart item would work when editing. Right now that matching pool amount is just calculated and shown at the bottom. Would we want to auto calculate and populate the cart item if we added a cart item for the matching pool? I can maybe ask some of these questions tomorrow in scrum incase these don't make sense.

**Comment by ryancreatescopy** [comment]
@corwintines I was imagining something like this:

If you reallocated to the matching pool, your cart will look like this in the tallying phase:
![image](https://user-images.githubusercontent.com/40891631/120766609-4f924000-c512-11eb-8fbd-a32613f2a137.png)

If you're editing your cart during reallocation, e.g. you reallocate once to give some to the matching pool and then you change your mind again and want to adjust the amounts:

![image](https://user-images.githubusercontent.com/40891631/120766723-6c2e7800-c512-11eb-8ed9-5af96fa48772.png)

I don't think it works to just treat it like the "Matching pool total" that you mentioned above because it needs to be clear you can edit the amount. 

We can use the round image to make it clearer it's the matching pool.

Some things to consider:
- do we need a warning that your contribution won't show in the matching pool total (in the round section) until after reallocation?
- if a user donates to the matching pool outside of reallocation (as a user you're able to add to the matching pool outside of your 1 contribution), should we show that in the cart too? This will complicate things somewhat as that amount won't factor into your cart total (e.g. amount you can reallocate). 
  - Perhaps with contributions to the matching pool outside of reallocation could be displayed in the matching pool component in the round info? Something like the rough sketch below?

![image](https://user-images.githubusercontent.com/40891631/120771092-c03b5b80-c516-11eb-980e-a4f7ff96439f.png)

  
cc @samajammin @wackerow @pettinarip some group thinking might be helpful on these questions

**Comment by ryancreatescopy** [comment]
> @ryancreatescopy couple questions:
> 
> * Do we want to display this during reallocation phase for user as well?

So my first thought... see comment above. If the user has reallocated and their contribution now includes the matching pool, if the user goes to reallocate again the matching pool amount should be easily configurable (e.g. change amount / remove etc.)

However now that I think a little more about it... if I reduce my matching pool amount in the UI, the remainder will default be going into the matching pool until I up the amounts for projects.  E.g. I remove 5 Dai from my 10 Dai MP contribution, that 5 Dai will channel back into the matching pool...

Perhaps during reallocation (e.g. editing cart) we only display matching pool in the totals.

> * Do we display a cart item for the matching pool before the tx? Or only one that reallocation has been made?

I think we only show it as an item in the read-only state once you've made the transaction. I think the totals (pictured below) make it clear enough to start 

![image](https://user-images.githubusercontent.com/40891631/120767566-33db6980-c513-11eb-9c2a-bf684b27b0d8.png)

**Comment by ryancreatescopy** [comment]
Ok, keeping my original thinking in comments above so that we can discuss them but here's maybe how it should work more clearly:

![image](https://user-images.githubusercontent.com/40891631/120769210-cb8d8780-c514-11eb-9391-c4d86b8aaabc.png)

In essence:
- matching pool contributions are shown as cart items in read-only state
- in edit mode we use the Matching pool total to explain how much is going to the matching pool due to the weirdness of editing the amount of the matching pool (as a cart item) and the remainder being sent back to the matching pool

**Comment by wackerow** [comment]
My thoughts on this:

If we're going to require some amount of contribution to the matching pool for any contributor, it feels more appropriate to break this line item off from the "cart items". 

User in **contribution** phase:
Once user has added an item to their cart, a component could be displayed just below the `CartItems` component that has a minimum value of whatever ends up being set (ie. 1 xDai)... So if user adds one project for 10 xDai, the total would show 11 xDai (if the minimum contribution to the matching pool was set to 1). 

- User should not be able to edit this. After chatting this through with @samajammin a bit, anyone trying to exclusively donate to the matching pool should be allowed to do so without any BrightID requirements, as these funds are not considered "votes" for the quadratic calculations. Given this, it makes more sense to me to _not_ allow user to edit the "matching pool funds" within the cart, and to only have that display the minimum required. If user wants to contribute more, exclusively to the matching pool, they should be allowed to do that separately without BrightID issues. 

- I do not see it fitting within the existing cart items list, since this basically holds a "Project" object (which the matching pool is not a "Project") along with an amount to contribute, plus a boolean to declare if it's been removed from the cart.
- By not including this in the array of cart items, this should prevent issues with "remove all"

Then, when user in **reallocation** phase: The total contribution/votes is set, and it seems simpler to not allow people to adjust the matching pool quantity directly, but to always calculate that based on what the remaining funds, while enforcing the minimum still.

cc: @ryancreatescopy @pettinarip @corwintines

**Comment by samajammin** [comment]
Ya I suspect it'd be worth having a group discussion on this.

The more I've thought on this, the more I'm inclined to advocate that the matching pool amount should NOT be a cart item, given how different it is from a contribution towards a project.

> I don't think it works to just treat it like the "Matching pool total" that you mentioned above because it needs to be clear you can edit the amount.

But can the user ever edit the amount (or do we want the amount to be editable)? My understanding is that the matching pool contribution would always be auto-calculated depending on the way you're reallocating. If someone wants to contribute additional funds directly to the matching pool, I'd argue to keep that as a separate flow for simplicity sake (since it is a separate transaction).

So to summarize, I like the overall ideas in @ryancreatescopy's latest comment:
1) When the cart is in edit mode, we show it as a line item at the bottom.
2) When the cart is in read mode, we COULD show the matching pool as a cart item (like in that design), though I'd also be perfectly happy with keeping it similar to just a line item like in edit mode.

**Comment by ryancreatescopy** [comment]
@samajammin so something like this for (2)

**Reallocation read-only** 
![image](https://user-images.githubusercontent.com/40891631/121137289-698f9380-c82e-11eb-8403-bc4d37d58c2d.png)

**Tallying read-only**
![image](https://user-images.githubusercontent.com/40891631/121137319-71e7ce80-c82e-11eb-9e73-cfe9465fa14e.png)

**Comment by samajammin** [comment]
Ya, I dig that @ryancreatescopy 👍 - keeps it simple (& makes it clear it's not an editable item).

**Comment by samajammin** [comment]
Solved in #154
